tests/e2e/page-utils.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file Generic utility functions for Puppeteer
+ */
+
+/**
+ * @typedef {import('@types/puppeteer').Page} Page
+ * @typedef {import('@types/puppeteer').ElementHandle} ElementHandle
+ * @typedef {import('@wordpress/blocks').Block} WPBlock
+ */
+
+/**
+ * Checks whether an element exists under a certain context
+ *
+ * @param {string} selector The selector for the desired element
+ * @param {Page | ElementHandle} [root=page] The root from which to search for the selector
+ *
+ * @return {Promise<boolean>} Whether the element exists or not
+ */
+export async function elementExists( selector, root = page ) {
```

Comment:
We could use instead of this the [toMatchElement](https://www.npmjs.com/package/expect-puppeteer#toMatchElement) utility that we already have via expect-puppeteer.

```
Expect an element be present in the page or element.
```

@sunyatasattva @tomasztunik

Reply:
@ralucaStan Oh but they are not mutually exclusive. `toMatchElement` is an assertion, this is not supposed to be an assertion. It can be used for example as a guard clause. Or you can see how it is used [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/035838f750c89bc3045bd723e53ed61cac283de3/tests/e2e/utils.js#L200).

Reply:
I see, thank you for explaining it to me. The example helped. Maybe it would be good to add a few words about the purpose, so that we don't end up using it in places where [toMatchElement](https://www.npmjs.com/package/expect-puppeteer#toMatchElement) might be better. What do you think?

Reply:
Yes! I agree, I didn't think about it, I'll add it to my notes.